adbc.h

```diff
@@ -459,6 +459,24 @@ const struct AdbcError* AdbcErrorFromArrayStream(struct ArrowArrayStream* stream
 ///
 /// \see AdbcConnectionGetInfo
 #define ADBC_INFO_VENDOR_ARROW_VERSION 2
+/// \brief Indicates whether SQL queries are supported (type: bool).
+///
+/// \see AdbcConnectionGetInfo
+#define ADBC_INFO_VENDOR_SQL 3
+/// \brief Indicates whether Substrait queries are supported (type: bool).
+///
+/// \see AdbcConnectionGetInfo
+#define ADBC_INFO_VENDOR_SUBSTRAIT 4
+/// \brief The minimum supported Substrait version, or null if
+///   Substrait is not supported (type: utf8).
```

Comment:
Would using a type other than utf8 for a version make sense?

The ADBC version appears to have a numeric encoding:

```c
/// \since ADBC API revision 1.1.0
#define ADBC_VERSION_1_1_0 1001000
```

Reply:
I think a numeric encoding would have some benefits, but in this case utf8 is most consistent with existing encodings of substrait version.

For context the version field of `SubstraitPlan` is a `string` in [FlightSql.proto](https://github.com/apache/arrow/blob/6aa33210ecd9773f18ceb775306b79de132b6c6f/format/FlightSql.proto#L1508). In substrait itself `Version` is a message that can either use numeric major.minor.patch versions or simply be a [git_hash](https://github.com/substrait-io/substrait/blob/b663a4aee64e6237566a1b930549d90b909e322e/proto/substrait/plan.proto#L74), so `string` may be required.

Reply:
Leaving a string makes sense to me then 👍